lib/ansible/modules/cloud/vultr/vr_sshkey_facts.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# (c) 2018, Yanis Guenane <yanis+ansible@guenane.org>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: vr_sshkey_facts
+short_description: Gather facts about the Vultr SSH keys available.
+description:
+  - Gather facts about SSH keys available.
+version_added: "2.7"
+author: "Yanis Guenane (@Spredzy)"
+extends_documentation_fragment: vultr
+'''
+
+EXAMPLES = r'''
+- name: Gather Vultr SSH keys facts
+  local_action:
+    module: vr_sshkey_facts
+
+- name: Print the gathered facts
+  debug:
+    var: ansible_facts.vultr_sshkey_facts
+'''
+
+RETURN = r'''
+---
+vultr_api:
+  description: Response from Vultr API with a few additions/modification
+  returned: success
+  type: complex
+  contains:
+    api_account:
+      description: Account used in the ini file to select the key
+      returned: success
+      type: string
+      sample: default
+    api_timeout:
+      description: Timeout used for the API requests
+      returned: success
+      type: int
+      sample: 60
+    api_retries:
+      description: Amount of max retries for the API requests
+      returned: success
+      type: int
+      sample: 5
+    api_endpoint:
+      description: Endpoint used for the API requests
+      returned: success
+      type: string
+      sample: "https://api.vultr.com"
+ansible_facts:
+  description: Response from Vultr API
+  returned: success
+  type: complex
+  contains:
+    "vultr_sshkey_facts": {
+      "me@home": {
+        "SSHKEYID": "5abf426403479",
```

Comment:
I would like to suggest to streamline the keys with the current modules. vultr is not very consistent in the keys and types of returns. Sometimes an id is a string, sometimes a int. etc.

The return should look like:

~~~json
"vultr_sshkey_facts": {
  "id": "5abf426403479",
  "date_created": "2018-02-24 15:04:01",
  "name": "me@home",
  "ssh_key": "ssh-rsa AAAAB3Nz...NnPz me@home"
}
~~~